src/Strategy/Python/Poetry/PyProject.hs

```diff
@@ -0,0 +1,311 @@
+toDependencyVersion dt = case parse parseConstraintExpr "" dt of
+  Left _ -> Nothing
+  Right v -> case v of
+    CEq "*" -> Nothing
+    vc -> Just vc
+
+toDependnecy :: [DepEnvironment] -> Map Text PoetryDependency -> Map Text Dependency
+toDependnecy depEnvs = M.mapWithKey $ poetryDependencyToDependency depEnvs
+
+getDependencies :: PyProject -> [Dependency]
+getDependencies project = filter ignorePython $ map snd allDeps
+  where
+    -- pyproject typically includes python as dependency that has to be ignored
+    ignorePython = (\d -> dependencyName d /= "python")
+    allDeps = case pyprojectPoetry project of
+      Nothing -> []
+      Just pp -> M.toList (prodDeps) ++ M.toList (devDeps)
+        where
+          prodDeps = toDependnecy [EnvProduction] (dependencies pp)
+          devDeps = toDependnecy [EnvDevelopment] (devDependencies pp)
+
+type Parser = Parsec Void Text
+
+symbol :: Text -> Parser Text
+symbol = L.symbol sc
+  where
+    sc :: Parser ()
+    sc = L.space (void $ some $ char ' ') empty empty
+
+-- | Consumes whitespace ` ` or tab `\t`
+whitespaceOrTab :: Parser Text
+whitespaceOrTab = takeWhileP (Just "whitespaceOrTab") isSpaceOrTab
+  where
+    isSpaceOrTab c = c == ' ' || c == '\t'
+
+-- | Poetry constraint operators
+data PoetryConstraintOperators
+  = GreaterThanOrEqual
+  | GreaterThan
+  | LessThanOrEqual
+  | LessThan
+  | Equal
+  | NotEqual
+  | WildcardAny
+  | MajorCompatible
+  | MinorCompatible
+
+-- | Parses `VerConstraint`
+parseVerConstraint :: Parser VerConstraint
+parseVerConstraint = do
+  operator <- whitespaceOrTab *> parseConstraintOperator <* whitespaceOrTab
+  versionText <- many (noneOf [andOperatorChar, orOperatorChar] <* whitespaceOrTab)
+  case operator of
+    (Equal) -> return $ CEq $ fromCharToText versionText
+    (NotEqual) -> return $ CNot $ fromCharToText versionText
+    (GreaterThanOrEqual) -> return $ CGreaterOrEq $ fromCharToText versionText
+    (GreaterThan) -> return $ CGreater $ fromCharToText versionText
+    (LessThanOrEqual) -> return $ CLessOrEq $ fromCharToText versionText
+    (LessThan) -> return $ CLess $ fromCharToText versionText
+    (MajorCompatible) -> return $ CCompatible $ fromCharToText versionText
+    (MinorCompatible) -> return $ CCompatible $ fromCharToText versionText
+    (WildcardAny) -> return $ CEq "*"
+  where
+    fromCharToText :: [Char] -> Text
+    fromCharToText chars = T.pack chars
+    andOperatorChar = ','
+    orOperatorChar = '|'
+
+-- | Parse poetry constraint operator
+parseConstraintOperator :: Parser PoetryConstraintOperators
+parseConstraintOperator = fromMaybe Equal <$> optional (asum (map symbol operatorList) >>= textToPoetryVersion)
+  where
+    operatorList = [">=", "<=", "~=", ">", "<", "^", "==", "===", "!=", "~", "*", "="] :: [Text]
+
+-- | Constructs `PoetryConstraintOperators` from textual representation (==, >=, etc.) of operators.
+textToPoetryVersion :: (MonadFail m) => Text -> m PoetryConstraintOperators
```

Comment:
Since you're only ever using this in one way, just concretize the monad to `Parser`.

Reply:
Why should we monomorphize here?  The compiler will generate the same code either way, and it makes much more clear to the reader that we **aren't** doing any parsing in this function.  The only thing we're doing is converting (not using the parser), and that either succeeds or fails.

Reply:
...eh, yeah, okay. I prefer to monomorphize to signal intent, but I'm okay either way. Maybe `where`-block it?